src/edu/stanford/nlp/sempre/InfiniteListValue.java

```diff
@@ -0,0 +1,66 @@
+package edu.stanford.nlp.sempre;
+
+import java.util.*;
+
+import fig.basic.LispTree;
+
+/**
+ * Represent a list of infinitely many values.
+ *
+ * The list is represented by a List of Objects.
+ *
+ * @author ppasupat
+ */
+public class InfiniteListValue extends Value {
```

Comment:
I don't see how this can represent an infinite list because we need to serialize and deseralize LispTrees, which have to be finite...